tests/integrationv2/test_jdk_happy_path.py

```diff
@@ -0,0 +1,49 @@
+import copy
+import pytest
+
+from configuration import available_ports
+from common import ProviderOptions, Protocols, Certificates, data_bytes
+from fixtures import managed_process
+from providers import Provider, S2N, JavaSSL
+from utils import invalid_test_parameters, get_parameter_name, get_expected_s2n_version
+
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", [Certificates.ECDSA_384], ids=get_parameter_name)
+def test_s2n_server_happy_path(managed_process, certificate, protocol):
```

Comment:
I'm still looking into this, but there are specific quirks about the jdk that make it not fit within our current testing framework. (See the pkcs 12 vs. pem format for certs discussion above) I don't know whether other providers also have peculiarities we'd have to account for.

Reply:
This could be addressed with some meta information in the Certificates struct. When we [initialize a certificate](https://github.com/awslabs/s2n/blob/main/tests/integrationv2/common.py#L74-L78), we could default to pem, but allow a different format. The JavaProvider could `pytest.skip` unsupported certificates formats. Or a kludgy way could be updating [invalid_test_parameters](https://github.com/awslabs/s2n/blob/main/tests/integrationv2/utils.py#L38) to invalidate certificates that are not supported with the JavaProvider.

Any tests that have certificate names hardcoded are not correct. I know some of those have snuck in so we can get tests working, but now its time to update the Certificates structs to make them correct. Those tests should be updated.

Reply:
I agree with Ryan. We should try to make the providers as interchangeable as possible. We don't want multiple copies of the same test for each provider.

Reply:
Another possibility is to build the logic into Java to handle PEM certificates. It's a tad more complicated but still doable. (I don't think it should block this PR though.)